system/lib/wasmfs/file.h

```diff
@@ -59,6 +63,32 @@ class File : public std::enable_shared_from_this<File> {
 
   public:
     Handle(std::shared_ptr<File> file) : file(file), lock(file->mutex) {}
+
+    void getStat(struct stat* buf) {
```

Comment:
It would be good to make only the system call layer know about `struct stat` and all the things inside it.